packages/dds/tree/src/test/domains/json/jsonCursorLegacy.bench.ts

```diff
@@ -0,0 +1,130 @@
+// equivalent clone of the source data.
+function bench(name: string, getJson: () => any) {
+    const json = getJson();
+    const encodedTree = jsonableTreeFromCursor(new JsonCursor(json));
+    const schemaData: SchemaData = {
+            globalFieldSchema: new Map(),
+            treeSchema: jsonTypeSchema,
+    };
+    const schema = new StoredSchemaRepository(defaultSchemaPolicy, schemaData);
+
+    benchmark({
+        type: BenchmarkType.Measurement,
+        title: `Direct: '${name}'`,
+        before: () => {
+            const cloned = clone(json);
+            assert.deepEqual(cloned, json,
+                "clone() must return an equivalent tree.");
+            assert.notEqual(cloned, json,
+                "clone() must not return the same tree instance.");
+        },
+        benchmarkFn: () => {
+            clone(json);
+        },
+    });
+
+    const cursorFactories: [string, () => ITreeCursor][] = [
+        ["JsonCursor", () => new JsonCursor(json)],
+        ["TextCursor", () => singleTextCursor(encodedTree)],
+        ["object-forest Cursor", () => {
+            const forest = buildForest(schema);
+            initializeForest(forest, [encodedTree]);
+            const cursor = forest.allocateCursor();
+            assert.equal(forest.tryMoveCursorTo(forest.root(forest.rootField), cursor), TreeNavigationResult.Ok);
+            return cursor;
+        }],
+    ];
+
+    const consumers: [string, (cursor: ITreeCursor) => void][] = [
+        ["cursorToJsonObject", cursorToJsonObject],
+        ["jsonableTreeFromCursor", jsonableTreeFromCursor],
+        ["sum", sum],
+    ];
+
+    for (const [consumerName, consumer] of consumers) {
+        for (const [factoryName, factory] of cursorFactories) {
+            let cursor: ITreeCursor;
+            benchmark({
+                type: BenchmarkType.Measurement,
+                title: `${consumerName}(${factoryName}): '${name}'`,
+                before: () => {
+                    cursor = factory();
+                    assert.deepEqual(cursorToJsonObject(cursor), json, "data should round trip through json");
+                    assert.deepEqual(
+                        jsonableTreeFromCursor(cursor), encodedTree, "data should round trip through jsonable");
+                },
+                benchmarkFn: () => {
+                    consumer(cursor);
+                },
+            });
+        }
+    }
+}
+
+function sum(cursor: ITreeCursor): number {
+    let total = 0;
+    const value = cursor.value;
+    if (typeof value === "number") {
+        total += value;
+    }
+    for (const key of cursor.keys) {
+        total += reduceField(cursor, key, total, sum);
+    }
+    return total;
+}
+
+const canada = generateCanada(
+    // Use the default (large) data set for benchmarking, otherwise use a small dataset.
+    isInPerformanceTestingMode
+        ? undefined
+        : [2, 10]);
+
+describe("ITreeCursor Legacy", () => {
+    bench("canada", () => canada);
```

Comment:
maybe pass in consumers, so they can be dataset specific

Reply:
Done